### PR TITLE
Various compatibility fixes and added tests against them

### PR DESCRIPTION
**Pull request by white238**
* Guard test that requires separable compilation
* Add new blueos c++17 host-config
* Remove BGQ host-config
* Guard system includes on PGI compilers
* Add ability to exclude CUDA implicit link dirs

I also reenabled (and hopefully fixed) our internal testing.  I am not sure who disabled it but it was on the wrong branch (main/master) and throwing alarming looking git errors.  I changed to a straight git repository.

### CODE REVIEW

File: host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_nvcc_c++17.cmake [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2020, Lawrence Livermore National Security, LLC and
+# other BLT Project Developers. See the top-level COPYRIGHT file for details
+# 
+# SPDX-License-Identifier: (BSD-3-Clause)
+
+#------------------------------------------------------------------------------
+# Example host-config file for the blue_os cluster at LLNL, specifically Lassen
```
Comment (kennyweiss) [inline host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_nvcc_c++17.cmake:7]: >specifically Lassen

Is this last clause necessary?

File: host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_nvcc_c++17.cmake [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#------------------------------------------------------------------------------
+#
+# This file provides CMake with paths / details for:
+#  C/C++:   Clang with GCC 8.3.1 toolchain
```
Comment (kennyweiss) [inline host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_nvcc_c++17.cmake:11]: Should this line mention `C++17`?

(I realize it is in the file name)

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -9,6 +9,11 @@ The project release numbers follow [Semantic Versioning](http://semver.org/spec/
 
 ## [Unreleased] - Release date yyyy-mm-dd
 
+## Added
+- Added variable to allow removing of implicit link directories added to CUDA executables
```
Comment (kennyweiss) [inline RELEASE-NOTES.md:13]: Reword suggestion:

>  - Added variable `BLT_CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES_EXCLUDE` to allow removing 
implicit link directories added to CUDA executables by CMake. See the following example host-config:
  `host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_nvcc_c++17.cmake`.

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -18,6 +23,9 @@ The project release numbers follow [Semantic Versioning](http://semver.org/spec/
 - ClangFormat checks now support multiple Python executable names
 - Prefixed blt_register_library() internal variables with ``_`` to avoid collision
   with input parameters.
+- Turn off system includes for registered libraries when using the PGI compiler
+- Removed unneeded INTERFACE system includes added by googletest that was causing problems
+  PGI builds (BLT was adding them already to the register library calls)
```
Comment (kennyweiss) [inline RELEASE-NOTES.md:28]: Suggestion (missing word): 
> ... causing problems \*in\* PGI builds...

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -18,6 +23,9 @@ The project release numbers follow [Semantic Versioning](http://semver.org/spec/
 - ClangFormat checks now support multiple Python executable names
 - Prefixed blt_register_library() internal variables with ``_`` to avoid collision
   with input parameters.
+- Turn off system includes for registered libraries when using the PGI compiler
```
Comment (davidbeckingsale) [inline RELEASE-NOTES.md:26]: Does PGI not understand this option?

Comment (white238) [reply RELEASE-NOTES.md:26]: Apparently not:

```
[white238@rzansel61:temp]$ /usr/tce/packages/pgi/pgi-20.4/bin/pgc++ -isystem include helloworldheader.cpp
pgc++-Error-Unknown switch: -isystem
[white238@rzansel61:temp]$ /usr/tce/packages/pgi/pgi-20.4/bin/pgc++ -I include helloworldheader.cpp
[white238@rzansel61:temp]$ ./a.out
Hello World!

```

**Pull request merged**